android/quest/src/main/java/org/smartregister/fhircore/quest/ui/launcher/GeoWidgetLauncherFragment.kt

```diff
@@ -16,125 +16,243 @@
+                val uiState: AppMainUiState = appMainViewModel.appMainUiState.value
+                val openDrawer: (Boolean) -> Unit = { open: Boolean ->
+                    scope.launch {
+                        if (open) scaffoldState.drawerState.open() else scaffoldState.drawerState.close()
+                    }
+                }
+
+                // Close side menu (drawer) when activity is not in foreground
+                val lifecycleEvent = rememberLifecycleEvent()
+                LaunchedEffect(lifecycleEvent) {
+                    if (lifecycleEvent == Lifecycle.Event.ON_PAUSE) scaffoldState.drawerState.close()
+                }
+
+                LaunchedEffect(Unit) {
+                    geoWidgetLauncherViewModel.snackBarStateFlow.hookSnackBar(
+                        scaffoldState = scaffoldState,
+                        resourceData = null,
+                        navController = findNavController(),
+                    )
+                }
+
+                AppTheme {
+                    // Register screen provides access to the side navigation
+                    Scaffold(
+                        drawerGesturesEnabled = scaffoldState.drawerState.isOpen,
+                        scaffoldState = scaffoldState,
+                        drawerContent = {
+                            AppDrawer(
+                                appUiState = uiState,
+                                openDrawer = openDrawer,
+                                onSideMenuClick = appMainViewModel::onEvent,
+                                navController = findNavController(),
+                            )
+
+                        },
+                        snackbarHost = { snackBarHostState ->
+                            SnackBarMessage(
+                                snackBarHostState = snackBarHostState,
+                                backgroundColorHex = appConfig.snackBarTheme.backgroundColor,
+                                actionColorHex = appConfig.snackBarTheme.actionTextColor,
+                                contentColorHex = appConfig.snackBarTheme.messageTextColor,
+                            )
+                        },
+                    ) { innerPadding ->
+                        Box(modifier = Modifier.padding(innerPadding) ) {
+
+                            val fragment = remember { geoWidgetFragment }
+
+                            GeoWidgetLauncherScreen(
+                                openDrawer = openDrawer,
+                                onEvent = geoWidgetLauncherViewModel::onEvent,
+                                navController = findNavController(),
+                                toolBarHomeNavigation = args.toolBarHomeNavigation,
+                                modifier = Modifier.fillMaxSize(), // Adjust the modifier as needed
+                                fragmentManager = childFragmentManager ,
+                                fragment = fragment
+                            )
```

Comment:
We may have to rethink this implementation to avoid unnecessary duplication. We could use navigation arguments to know when to switch between the `RegisterFragment` and this `GeowidgetFragmentLauncher` that way will avoid duplicating code from the register fragment.

Reply:
I rethink of the implemention and yeah, we needed this because GeoWidgetLauncherScreen has its own code of retrieving locations, processing them etc. I have other PR where I did some work. We can sync up on this to discuss